### PR TITLE
[systemtest] Add tests with mounting secrets and CM as volumes to ConnectS2I and improve the base test with doted names

### PR DESCRIPTION
**Pull request by im-konge**
Signed-off-by: Lukas Kral <lukywill16@gmail.com>

### Type of change

- New test
- New checks

### Description

After #3820 where the names of the volumes/config maps and secrets was fixed, we are able to name them with doted names (like `my.secret` or `my.config.map`). This PR also adds test which we had only in Connect also to `ConnectS2IST` to have it properly covered.

### Checklist

- [x] Write tests
- [ ] Make sure all tests pass

### CODE REVIEW

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ConnectS2IST#testMountingSecretAndConfigMapAsVolumesAndEnvVars,ConnectST#testMountingSecretAndConfigMapAsVolumesAndEnvVars

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectS2IST.java [hunk 1]
```diff
@@ -739,6 +746,158 @@ void testHostAliases() {
...
+
+        String dotedConfigMapVolumeName = "connect.config.map";
+        String dotedSecretVolumeName = "connect.secret";
+
+        String configMapKey = "my-key";
+        String secretKey = "my-secret-key";
+
+        Secret connectSecret = new SecretBuilder()
+            .withNewMetadata()
+            .withName("my-secret")
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectS2IST.java:772]: Maybe put name into var?

Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectS2IST.java:772]: Indent

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectS2IST.java [hunk 2]
```diff
@@ -739,6 +746,158 @@ void testHostAliases() {
...
+            .endMetadata()
+            .withType("Opaque")
+            .addToData(secretKey, encodedPassword)
+            .build();
+
+        ConfigMap configMap = new ConfigMapBuilder()
+            .editOrNewMetadata()
+            .withName("my-config-map")
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectS2IST.java:780]: Maybe put name into var?

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectS2IST.java [hunk 3]
```diff
@@ -739,6 +746,158 @@ void testHostAliases() {
...
+            .endMetadata()
+            .addToData(configMapKey, "my-value")
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectS2IST.java:782]: Maybe put value into var?

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java [hunk 4]
```diff
@@ -1031,16 +1031,23 @@ void testScaleConnectAndConnectorSubresource() {
...
 
+        String dotedSecretEnv = "MY_DOTED_CONNECTOR_SECRET";
+        String dotedConfigMapEnv = "MY_DOTED_CONNECT_CONFIG_MAP";
+
         String configMapVolumeName = "connect-config-map";
         String secretVolumeName = "connect-secret";
 
+        String dotedConfigMapVolumeName = "connect.config.map";
+        String dotedSecretVolumeName = "connect.secret";
+
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java:1050]: Maybe we should put this into some method? It's the same for ConnectST and ConnectS2IST. Same with creation of serets/cm

**Review by Frawless — COMMENTED** [review]
Shouldn't we check it for mm2 as well?

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectS2IST.java (hunk 1 above)
Comment (Frawless) [reply systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectS2IST.java:772]: It's in multiple places, I am list lazy mark every line :)

**Review by scholzj — APPROVED** [review]

**Comment by im-konge** [comment]
@Frawless I asked @scholz before about the mm etc. and it seems that Connect/ConnectS2I is enough. If not, I can just simply add it to mm/mm2 suite.

**Comment by Frawless** [comment]
Only mm2 use Connect, but if you already ask, it should be fine.

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java (hunk 4 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java:1050]: Well the tests are practically same, only difference is between the CRs -> I would like to make the tests "more generic" how we discussed offline. But for now, I guess, I can create some method in `AbstractST` and then just refactor it.
WDYT @Frawless ?

Comment (Frawless) [reply systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java:1050]: Discussed offline

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ConnectS2IST#testMountingSecretAndConfigMapAsVolumesAndEnvVars,ConnectST#testMountingSecretAndConfigMapAsVolumesAndEnvVars

**Review by Frawless — APPROVED** [review]

**Pull request merged**